src/libYARP_dev/src/devices/JoypadControlClient/JoypadControlClient.h

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (C) 2017 iCub Facility - Istituto Italiano di Tecnologia
+ * Author: Andrea Ruzzenenti <andrea.ruzzenenti@iit.it>
+ * CopyPolicy: Released under the terms of the LGPLv2.1 or later, see LGPL.TXT
+ */
+
+#include <yarp/dev/IJoypadController.h>
+#include <yarp/dev/DeviceDriver.h>
+#include <yarp/os/RateThread.h>
+#include <vector>
+#include "../JoypadControlServer/JoypadControlNetUtils.h"
+
+#define DEFAULT_THREAD_PERIOD 10
+namespace yarp
+{
+    namespace dev
+    {
+        class JoypadControlClient;
+    }
+}
+
+class yarp::dev::JoypadControlClient : public yarp::dev::IJoypadController,
+                                       public yarp::dev::DeviceDriver
+{
+private:
+    //---------------utils
+    #define JOYPORT yarp::dev::JoypadControl::JoyPort
```

Comment:
i think i can't use typedef here since it's a template class and i think it's better to leave the type in the class's member variable declaration instead of declaring multiple typedef (one per type). also there is an undef at the end of the class in order to keep the things safe and clean. 

Any suggestion for getting rid of the "define" but keeping the templateType specification per member and at the same time mantaining a short type declaration?